minio-client/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	minio "github.com/minio/minio-go"
+)
+
+var (
+	accessKeyID              = os.Getenv("ACCESS_KEY")
+	secretAccessKey          = os.Getenv("SECRET_KEY")
+	serviceName              = os.Getenv("SERVICE_NAME")
+	namespace                = os.Getenv("NAMESPACE")
+	port                     = os.Getenv("PORT")
+	livenessCheckInterval, _ = strconv.Atoi(os.Getenv("LIVENESS_CHECK_INTERVAL"))
+	livenessRetryCount, _    = strconv.Atoi(os.Getenv("LIVENESS_RETRY_COUNT"))
+	mode                     = os.Getenv("MODE")
+	capacity, _              = strconv.ParseFloat(os.Getenv("CAPACITY"), 32)
+)
+
+func main() {
+	fileName := "bigfile.test"
+	bucketName := "test-bucket" + "-" + fmt.Sprintf("%d", time.Now().Unix())
+	endpoint := serviceName + "." + namespace + ".svc.cluster.local:" + port
+	useSSL := false
+
+	// Initialize minio client object.
+	minioClient, err := minio.New(endpoint, accessKeyID, secretAccessKey, useSSL)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// log.Printf("%#v\n", minioClient) // minioClient is now setup
+
+	createBucket(minioClient, bucketName, "us-east-1")
+	if mode == "liveness" {
+		createContent(fileName, 2097152)
+		livenessCheck(minioClient, bucketName, fileName)
+	} else {
+		fileSize := int(capacity * 1024 * 1024 * 1024)
+		createContent(fileName, fileSize)
+		fmt.Println("Load generation may take some time")
+		loadBucket(minioClient, bucketName, fileName)
+	}
+
+}
+
+func createBucket(client *minio.Client, bucketName string, region string) {
+	err := client.MakeBucket(bucketName, region)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	fmt.Println("Successfully Created bucket with name: ", bucketName)
+
+}
+
+func createContent(fileName string, fileSize int) {
+	bigBuff := make([]byte, fileSize)
+	ioutil.WriteFile(fileName, bigBuff, 0666)
+}
+
+func loadBucket(client *minio.Client, bucketName string, fileName string) error {
+	file, err := os.Open(fileName)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	defer file.Close()
+
+	fileStat, err := file.Stat()
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	n, err := client.PutObject(bucketName, "test-object", file, fileStat.Size(), minio.PutObjectOptions{ContentType: "application/text"})
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	fmt.Println("Successfully uploaded bytes: ", n)
+	return nil
+}
+
+func unloadBucket(client *minio.Client, bucketName string) error {
+	err := client.RemoveObject(bucketName, "test-object")
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	return nil
+}
+
+func livenessCheck(client *minio.Client, bucketName string, fileName string) {
+	count := 0
+	for {
+		if err := loadBucket(client, bucketName, fileName); err != nil {
+			count++
+			fmt.Println("Liveness failed, Retrying...")
+			if count > livenessRetryCount {
```

Comment:
typically, app toleration is measured against a timeout. Can we derive the count by taking a timeout instead?

Reply:
Do we plan to do this in a different PR? I still see retries based on a count. If so, could you please raise an issue to track this?